mbuild/packing.py

```diff
@@ -44,16 +44,50 @@
 end structure
 """
 
-PACKMOL_CONSTRAIN = """
-constrain_rotation x 0. 0.
-constrain_rotation y 0. 0.
-constrain_rotation z 0. 0.
-"""
+def packmol_constrain(fix_orientation):
+    """
+    Provides information to PACKMOL about which axes to constrain rotation.
+    fix_orientation is generated within the `fill_box` function
+    Parameters
+    ----------
+    fix_orientation : iterable of booleans, directions (x, y, z) about
+                      which to constrain rotation. If True, no rotation
+                      in that direction
+    Returns
+    -------
+    None if fix_orientation = (False, False, False)
+    string
+        rotation constraints to be added to PACKMOL input text
+    """
+    
+    constraints = ['constrain_rotation x 0. 0.',
+                   'constrain_rotation y 0. 0.',
+                   'constrain_rotation z 0. 0.']
+
+    CONSTRAIN = """
+    {}
+    {}
+    {}
+    """
+    if not any(fix_orientation): # No directions fixed
+        return None
+    if all(fix_orientation): # All directions fixed
+        PACKMOL_CONSTRAIN = CONSTRAIN.format(constraints[0], constraints[1], constraints[2])
+    elif fix_orientation.count(True) == 1: # Only 1 direction fixed
+        PACKMOL_CONSTRAIN = CONSTRAIN.format(constraints[fix_orientation.index(True)],"","")
+    else: # 2 directions are fixed
+        if fix_orientation[0] and fix_orientation[1]: # Only x and y are fixed
+            PACKMOL_CONSTRAIN = CONSTRAIN.format(constraints[0], constraints[1], "")
+        if fix_orientation[0] and fix_orientation[2]: # Only x and z are fixed
+            PACKMOL_CONSTRAIN = CONSTRAIN.format(constraints[0], constraints[2], "")
+        if fix_orientation[1] and fix_orientation[2]: # Only y and z are fixed
+            PACKMOL_CONSTRAIN = CONSTRAIN.format(constraints[1], constraints[2], "")
+    return PACKMOL_CONSTRAIN
```

Comment:
One other option that might be aligned with jenny's suggestion

```python3
final_constraints = list()
for i,val in enumerate(fix_orientation):
  if val:
    final_constraints.append(constraints[i])
  else:
    final_constraints.append("")
PACKMOL_CONSTRAIN = CONSTRAIN.format(final_constraints**)
```

Yours is more readable, but this removes all of the conditionals from line 74 onwards _i think_